tests/nd_item/nd_item_api.cpp

```diff
@@ -21,175 +21,136 @@
 
 #include "../common/common.h"
 
+#include <algorithm>
+
 #define TEST_NAME nd_item_api
 
 namespace test_nd_item__ {
 using namespace sycl_cts;
 
-size_t getIndex(sycl::id<1> Id, sycl::range<1> Range) {
-  return Id.get(0);
-}
+size_t getIndex(sycl::id<1> Id, sycl::range<1> Range) { return Id.get(0); }
 
 size_t getIndex(sycl::id<2> Id, sycl::range<2> Range) {
   return Id.get(1) + Id.get(0) * Range.get(1);
 }
 
 size_t getIndex(sycl::id<3> Id, sycl::range<3> Range) {
-  return Id.get(2) + Id.get(1) * Range.get(0) + Id.get(0) * Range.get(0) * Range.get(1);
+  return Id.get(2) + Id.get(1) * Range.get(0) +
+         Id.get(0) * Range.get(0) * Range.get(1);
 }
 
 template <int dimensions>
 class kernel_nd_item {
  protected:
-  typedef sycl::accessor<int, dimensions, sycl::access_mode::read,
-                             sycl::target::device>
-      t_readAccess;
-  typedef sycl::accessor<int, dimensions, sycl::access_mode::write,
-                             sycl::target::device>
-      t_writeAccess;
+  using t_readAccess = sycl::accessor<int, dimensions, sycl::access_mode::read,
+                                      sycl::target::device>;
+  using t_writeAccess =
+      sycl::accessor<int, dimensions, sycl::access_mode::write,
+                     sycl::target::device>;
 
   t_readAccess m_globalID;
   t_readAccess m_localID;
-  t_writeAccess m_o;
+  t_writeAccess m_out;
+  t_writeAccess m_out_deprecated;
 
  public:
-  kernel_nd_item(t_readAccess inG_, t_readAccess inL_, t_writeAccess out_)
-      : m_globalID(inG_), m_localID(inL_), m_o(out_) {}
+  kernel_nd_item(t_readAccess inG_, t_readAccess inL_, t_writeAccess out_,
+                 t_writeAccess out_deprecated_)
+      : m_globalID(inG_),
+        m_localID(inL_),
+        m_out(out_),
+        m_out_deprecated(out_deprecated_) {}
 
   void operator()(sycl::nd_item<dimensions> myitem) const {
-    bool failed = false;
+    bool all_correct = true;
```

Comment:
Can I whine a little? :)
What I don't like about this test is that the error report is not informative. The kernel tests quite a lot of member functions and only return a single bool status for all of them. When test fails it's not possible to identify what exactly doesn't work from the test output.
I'm not pushing for fixing this right away, but it's something we should add in our TODO list.

Reply:
I wholeheartedly agree with you, but for this PR I have tried to keep the changes minimal. For `nd_item_api.cpp` specificly, I made more changes than in the other files, because it used an `are_same` condition in the kernel, only to flip the condition before returning it. Now, it is at least more intuitive and equal to how it's done for `item`.

The problem you decribe exists for many more tests. An approach like the one taken in `group` or `sub_group` is much more suitable. I have created a follow-up issue here: https://github.com/KhronosGroup/SYCL-CTS/issues/498

A similar point is to update the old testing framework to the new one, for all tests. It's an quick and easy change, but results in a lot of modifications making it hard to review.